scripts/extensions/sams/src/store/sets/types.ts

```diff
@@ -0,0 +1,56 @@
+// Types
+import {CONTENT_PANEL_STATE, ISetItem} from '../../interfaces';
+
+export const RECEIVE_SETS = 'sets__receive';
```

Comment:
Is there a reason to export these constants instead of inlining them in the interface?

Reply:
This has already been covered in the link I provided: https://redux.js.org/recipes/usage-with-typescript#type-checking-actions--action-creators

Reply:
As I have stated earlier:
> If any developer wishes to learn how state management is used in this extension, we can point that developer to the Redux website and the above recipe provided.

The code works perfectly fine this way, I do not see the problem

Reply:
If you don't see the problem, but I see it, why not fix it? Or do you see a problem with not using constants?

The problem that I see is that we are adding additional lines of code that aren't needed. If the problem can be solved in 50 lines of code, why write 500 lines to solve it? Why create variables and manage imports when it can be avoided?

Reply:
You're over-exaggerating the boilerplate added by following the examples provided by the Redux team

Reply:
I am over-exaggerating, but that was just an example to illustrate the point. If there are lines that aren't really useful for anything and can be dropped without a negative impact on code quality, why not do so? Are we obligated to repeat patterns from documentation pages even if they aren't useful?

Let me know if I'm wrong and there is any benefit to having those constants.

Reply:
These constants are used in multiple places:
https://github.com/superdesk/superdesk-client-core/blob/61cafacbf209090496b108c5132368421d774469/scripts/extensions/sams/src/store/sets/types.ts#L6
https://github.com/superdesk/superdesk-client-core/blob/61cafacbf209090496b108c5132368421d774469/scripts/extensions/sams/src/store/sets/actions.ts#L21
https://github.com/superdesk/superdesk-client-core/blob/61cafacbf209090496b108c5132368421d774469/scripts/extensions/sams/src/store/sets/reducers.ts#L25

This is quite common practice in any programming language. If you think this code will cause bugs or will not work in any way, I will change it.

Reply:
Rain dances were also a common practice. We shouldn't be copying things without understanding them just because others do it. That's the main issue I'm concerned about much more than the constants themselves.

I'll help you out and link the reasons why people use constants for action names: https://stackoverflow.com/a/37054518/1175593

These are good points, but they are irrelevant when TypeScript is used because there already is a union of all possible actions in a single place that satisfies all listed goals.